[DOCS] Adds performance considerations section to transforms overview (#53791)
elastic/elasticsearch: docs/reference/transform/overview.asciidoc

IMPORTANT: The {transform} leaves your source index intact. It
creates a new index that is dedicated to the transformed data.

[[transform-performance]]

> ==== Performance considerations

Reviewer: I think we could make a case for putting this section in many places (e.g. Troubleshooting, Limitations, or in a whole new page).  I kind of expected it to be a new page, but given how short it is, I think this location makes sense.  If it grows later, I think we should reconsider.

Author: @lcawl Exactly my thoughts. I planned to add as an individual page, but it is very short for that. When we expand the use case, we move it to a new page.

> As mentioned earlier, {transforms} perform search aggregations on the source 

Reviewer: I don't think you need this introductory phrase, since folks might jump to this section
```suggestion
{transforms} perform search aggregations on the source 
```

> indices then index the results into the destination index. It means a 

Reviewer: Maybe it's just me, but starting a sentence with "it" always makes me a bit confused about what "it" is:
```suggestion
indices then index the results into the destination index. Therefore, a 
```

> {transform} never will take less time than the cumulated duration of the 

Reviewer: Removing what I think is unnecessary future  tense:
```suggestion
{transform} never takes less time than the cumulated duration of the 
```

aggregation that it performs and the indexing process.

For better performance, make sure that your search aggregations and queries are 

> optimized, so they don't process data that is not necessary for your 

Reviewer: I think this can be shortened:
```suggestion
optimized, so they don't process unnecessary data.
```

> {transform}.

Reviewer: ```suggestion

```
